service/catalog.go

```diff
@@ -84,16 +160,33 @@ func updateCatalog(w http.ResponseWriter, r *http.Request, envId string) (int, e
 		return http.StatusBadRequest, err
 	}
 
-	if err := db.Model(&model.CatalogModel{}).Where(&model.CatalogModel{
+	vars := mux.Vars(r)
+	oldCatalogName := vars["catalog"]
+
+	oldCatalog := model.CatalogModel{
 		Catalog: model.Catalog{
-			Name:          catalogModel.Name,
+			Name:          oldCatalogName,
 			EnvironmentId: envId,
 		},
-	}).Update(catalogModel).Error; err != nil {
+	}
+
+	if isDuplicateGlobalName(catalogModel) {
```

Comment:
So is DuplicateEnvName does a different match. https://github.com/rancher/catalog-service/pull/81/files#diff-b61cc85ce57f254e43ae79383f420dcaR75

The reason I add this is because this indicates that this is an update rather than a catalog trying to change to a duplicate name that exists in the environment already.